File: webv2/summary/summary.go
package summary

import "github.com/cloudspannerecosystem/harbourbridge/internal/reports"

type ConversionSummary struct {
	SrcTable         string
	SpTable          string
	Errors           []reports.WarningClassified
	Warnings         []reports.WarningClassified
	Suggestions      []reports.WarningClassified
	Notes            []reports.WarningClassified
	ErrorsCount      int
	WarningsCount    int
	SuggestionsCount int
	NotesCount       int
}

File: ui/src/app/model/summary.ts
export default interface ISummary {
  SrcTable: string
  SpTable: string
  Errors: WarningClassified[]
  Warnings: WarningClassified[]
  Suggestions: WarningClassified[]
  Notes: WarningClassified[]
  ErrorsCount: number
  WarningsCount: number
  SuggestionsCount: number
  NotesCount: number
}
export interface ISummaryRow {
  type: string
  content: string
  isRead: boolean
}

export interface WarningClassified {
	issueType: string
	description: string
}
